File: setup-py/create_all_indexes.py
"""Creates Moss indexes for all five COCO tiers sequentially.

Requires MOSS_PROJECT_ID, MOSS_PROJECT_KEY, and MOSS_INDEX_NAME in .env.
Data files (coco-data-*.json) must exist — run download_coco.py first.
"""

from __future__ import annotations

import asyncio

from dotenv import load_dotenv

from create_index import VALID_TIERS, create_index_for_tier

load_dotenv(".env")


async def create_all_indexes() -> None:
    """Create indexes for all tiers."""
    print("Creating indexes for all tiers...\n")

    for tier in VALID_TIERS:
        print(f"\n--- Tier: {tier} ---")
        try:
            await create_index_for_tier(tier)
        except Exception as error:
            print(f"Failed to create index for tier {tier}: {error}")

    print("\nAll indexes created.")


if __name__ == "__main__":
    asyncio.run(create_all_indexes())

File: setup-js/createAllIndexes.ts
/**
 * Creates Moss indexes for all four COCO tiers sequentially.
 *
 * Requires MOSS_PROJECT_ID, MOSS_PROJECT_KEY, and MOSS_INDEX_NAME in .env.
 * Data files (coco-data-*.json) must exist — run downloadCoco.ts first.
 */

import * as path from "path";
import { createIndexForTier, VALID_TIERS } from "./createIndex";
import { config } from "dotenv";

config({ path: path.resolve(__dirname, "../.env") });

async function createAllIndexes(): Promise<void> {
  console.log("Creating indexes for all tiers...\n");

  for (const tier of VALID_TIERS) {
    console.log(`\n--- Tier: ${tier} ---`);
    try {
      await createIndexForTier(tier);
    } catch (error) {
      console.error(`Failed to create index for tier ${tier}:`, error);
    }
  }

  console.log("\nAll indexes created.");
}

createAllIndexes().catch(console.error);
